Unmatched error type in std.os.send

### Zig Version

0.10.0-dev.4327+3a9344d8f

### Steps to Reproduce

Build below code with `zig build-exe`
```zig
const std = @import("std");
const x = std.x;

pub fn main() !void {
        var client = try x.net.tcp.Client.init(.ip, .{});
        defer client.deinit();
        const addr = x.net.ip.Address.initIPv4(x.os.IPv4.localhost, 8443);
        try client.connect(addr);
        _ = try client.write("ABC", 0);
}
```

### Expected Behavior

Build successes
```console
$ zig build-exe main.zig
$ echo $?
0
```

### Actual Behavior

Build fails
```console
$ zig build-exe main.zig
/home/naoki/.local/zig-master/lib/std/os.zig:5921:28: error: expected type 'error{AccessDenied,BrokenPipe,ConnectionResetByPeer,FastOpenAlreadyInProgress,FileDescriptorNotASocket,MessageTooBig,NetworkSubsystemFailed,NetworkUnreachable,SystemResources,Unexpected,WouldBlock}', found 'error{AccessDenied,BrokenPipe,ConnectionResetByPeer,FastOpenAlreadyInProgress,FileDescriptorNotASocket,MessageTooBig,NetworkSubsystemFailed,SystemResources,Unexpected,UnreachableAddress,WouldBlock}'
        else => |e| return e,
                           ^
/home/naoki/.local/zig-master/lib/std/os.zig:5921:28: note: 'error.UnreachableAddress' not a member of destination error set
referenced by:
    write: /home/naoki/.local/zig-master/lib/std/x/os/socket_posix.zig:69:22
    write: /home/naoki/.local/zig-master/lib/std/x/net/tcp.zig:137:27
    remaining reference traces hidden; use '-freference-trace' to see all reference traces
```

--- a/lib/std/os.zig
+++ b/lib/std/os.zig
@@ -5918,6 +5918,7 @@ pub fn send(
         error.NetworkUnreachable => unreachable,
         error.AddressNotAvailable => unreachable,
         error.SocketNotConnected => unreachable,
+        error.UnreachableAddress => unreachable,
         else => |e| return e,
     };
 }

